ppx/ppx.ml

```diff
@@ -510,30 +566,41 @@ let jsxMapper () =
           raise
             (Invalid_argument
                "A spread as a DOM element's children don't make sense written \
-                together. You can simply remove the spread.")
+                together. You can simply remove the spread." )
     in
     let args =
       match nonChildrenProps with
-      | [_justTheUnitArgumentAtEnd] ->
+      | [_justTheUnitArgumentAtTheEnd] ->
           [ (* "div" *)
             (nolabel, componentNameExpr)
           ; (* React.Dom.props(~className=blabla, ~foo=bar, ()) *)
-            (labelled "props", [%expr React.Dom.domProps ()])
+            (labelled "props", [%expr Js_of_ocaml.Js.Unsafe.obj [||]])
           ; (* [|moreCreateElementCallsHere|] *)
             (nolabel, childrenExpr) ]
       | nonEmptyProps ->
-          let propsCall =
-            Exp.apply ~loc
-              (Exp.ident ~loc
-                 {loc; txt= Ldot (Ldot (Lident "React", "Dom"), "domProps")})
-              ( nonEmptyProps
-              |> List.map (fun (label, expression) ->
-                     (label, mapper.expr mapper expression)) )
+          (* Filtering out the props that don't have a label, not sure how's possible *)
+          let propsWithName =
+            List.filter (fun (name, _) -> getLabel name != "") nonEmptyProps
+          in
+          let makePropField (arg_label, value) =
+            let name = getLabel arg_label in
+            let prop = Html.findByName name in
+            [%expr
+              [%e
+                Exp.constant ~loc (Pconst_string (Html.getHtmlName prop, None))]
+              (* loc here points to the element <div />, we could be more precise and point to the prop *)
```

Comment:
Maybe you can read `pexp_loc` from the label expression that is attached to each element in `nonEmptyProps`? 🤔

Reply:
It might require a non-trivial refactor, but will do that I believe those locations are worth every detail